pkg/kubelet/kubelet.go

```diff
@@ -1159,6 +1192,7 @@ func (kl *Kubelet) SyncPods(allPods []*api.Pod, podSyncTypes map[types.UID]metri
 	//      though the pod was not considered terminated by the apiserver).
 	// These two conditions could be alleviated by checkpointing kubelet.
 	pods := kl.filterOutTerminatedPods(allPods)
+	pods = kl.filterOutPodsPastActiveDeadline(pods)
```

Comment:
I am now of the opinion that the proper behavior is to move the ```handle***``` functions to after the filter call to stop checking pods that are failed or succeeded in this routine and having that potentially change the state of a succeeded pod as a host port conflict or as not fitting.  At least, that is how my current read of the code makes me think what is happening.  Please clarify if I am missing something.